crates/uv/src/commands/self_update.rs

```diff
@@ -0,0 +1,59 @@
+use std::fmt::Write;
+
+use anyhow::Result;
+use axoupdater::AxoUpdater;
+use owo_colors::OwoColorize;
+
+use crate::commands::ExitStatus;
+use crate::printer::Printer;
+
+/// Attempt to update the `uv` binary.
+pub(crate) async fn self_update(printer: Printer) -> Result<ExitStatus> {
+    let mut updater = AxoUpdater::new_for("uv");
+
+    // Load the "install receipt" for the current binary. If the receipt is found, then
+    // `uv` was likely installed via a package manager.
+    let Ok(receipt) = updater.load_receipt() else {
+        writeln!(
+            printer.stderr(),
+            "{}",
+            format_args!(
+                "{}{} Self-update is only available for `uv` binaries installed via the standalone installation scripts.\n\nIf you installed `uv` with `pip`, `brew`, or another package manager, update `uv` with `pip install --upgrade`, `brew upgrade`, or similar.",
+                "warning".yellow().bold(),
+                ":".bold()
+            )
+        )?;
+        return Ok(ExitStatus::Error);
+    };
+
+    if receipt.run().await? {
```

Comment:
```suggestion
    let was_updated = receipt.run().await.map_err(|err| {
        if let AxoupdateError::Reqwest(err) = err {
            anyhow::Error::new(BetterReqwestError::from(err))
        } else {
            anyhow::Error::new(err)
        }
    })?;
    if was_updated {
```

**before**

```
error: error sending request for url (https://api.github.com/repos/astral-sh/uv/releases): error trying to connect: dns error: failed to lookup address information: Temporary failure in name resolution
  Caused by: error trying to connect: dns error: failed to lookup address information: Temporary failure in name resolution
  Caused by: dns error: failed to lookup address information: Temporary failure in name resolution
  Caused by: failed to lookup address information: Temporary failure in name resolution
```

**after**

```
error: Could not connect, are you offline?
  Caused by: error sending request for url (https://api.github.com/repos/astral-sh/uv/releases): error trying to connect: dns error: failed to lookup address information: Temporary failure in name resolution
  Caused by: error trying to connect: dns error: failed to lookup address information: Temporary failure in name resolution
  Caused by: dns error: failed to lookup address information: Temporary failure in name resolution
  Caused by: failed to lookup address information: Temporary failure in name resolution
```

Still too verbose but clearer as to what went wrong.